### PR TITLE
Update auto-key, sorting, add some more utilities

### PR DESCRIPTION
**Pull request by BigYu**

### CODE REVIEW

File: src/projections/sorting.js [hunk 1]
```diff
@@ -13,13 +13,25 @@ export default function sorting({
           sorting: {
             $td = null,
             onSort = () => {},
-            reducer = (s, { key }) => (s === key ? null : key),
-            isSorting = ({ key }) => state === key,
+            reducer = (s, { key }) => {
```
Comment (wewei) [inline src/projections/sorting.js]: Do we really want the default reducer to have this complex behavior?

File: src/utils/object.js [hunk 2]
```diff
@@ -37,4 +37,17 @@ export const pick = (obj, ...keys) => {
     (searchKeys => searchKeys.indexOf(key) > -1)(flattendKeys));
 };
 
+export const omit = (obj, ...keys) => {
```
Comment (wewei) [inline src/utils/object.js]: not used

File: src/utils/object.js [hunk 3]
```diff
@@ -37,4 +37,17 @@ export const pick = (obj, ...keys) => {
...
+  const flattendKeys = flatten(keys);
+
+  return iterateeWithCondition(obj, (item, key) =>
+    (searchKeys => searchKeys.indexOf(key) === -1)(flattendKeys));
+};
+
 export const capitalizeFirstLetter = string => string.charAt(0).toUpperCase() + string.slice(1);
+
+export const updateKeys = (obj, keyReplacements) => Object.keys(obj).reduce((memo, key) => {
```
Comment (wewei) [inline src/utils/object.js]: not used

File: src/projections/sorting.js (hunk 1 above)
Comment (wewei) [reply src/projections/sorting.js]: If we want to provide the complex behavior, I'd rather to provide helper functions. Like

```js
import bidirectionalSortStrategy from '...';
{
  ...,
  sorting: bidirectionalSortStrategy({
    ...,
  }),
},
```

File: src/projections/auto-key.js [hunk 4]
```diff
@@ -1,16 +1,28 @@
...
+});
 
-export default autoKey;
+export default function ({ composeTable }) {
+  return {
+    composeTable(table) {
+      const model = composeTable(table);
+
+      return assign({}, model, {
+        colgroups: mapWithKey(model.colgroups),
```
Comment (wewei) [inline src/projections/auto-key.js]: and `cols` in `colgroup`.

File: src/utils/object.js (hunk 2 above)
Comment (BigYu) [reply src/utils/object.js]: Maybe we can refactor in the next iteration. Currently react grid uses it.

File: src/utils/object.js (hunk 3 above)
Comment (BigYu) [reply src/utils/object.js]: Maybe we can refactor in the next iteration. Currently react grid uses it.

**Pull request merged**